application/src/main/java/org/togetherjava/tjbot/commands/help/HelpSystemHelper.java

```diff
@@ -333,6 +333,55 @@ private void executeUncategorizedAdviceCheck(long threadChannelId, long authorId
         }).queue();
     }
 
+    void scheduleNoActivityAdviceCheck(long threadChannelId, long authorId) {
+        SERVICE.schedule(() -> {
+            try {
+                executeNoActivityAdviceCheck(threadChannelId, authorId);
+            } catch (Exception e) {
+                logger.warn(
+                        "Unknown error during a no activity advice check on thread {} by author {}.",
+                        threadChannelId, authorId, e);
+            }
+        }, SEND_NO_ACTIVITY_ADVICE_AFTER_MINUTES, TimeUnit.MINUTES);
+    }
+
+    private void executeNoActivityAdviceCheck(long threadChannelId, long authorId) {
+        logger.debug("Executing no activity advice check for thread {} by author {}.",
+                threadChannelId, authorId);
+
+        ThreadChannel threadChannel = jda.getThreadChannelById(threadChannelId);
+        if (threadChannel == null) {
+            logger.debug(
+                    "Channel for no activity advice check seems to be deleted (thread {} by author {}).",
+                    threadChannelId, authorId);
+            return;
+        }
+
+        if (threadChannel.isArchived()) {
+            logger.debug(
+                    "Channel for no activity advice check is archived already (thread {} by author {}).",
+                    threadChannelId, authorId);
+            return;
+        }
+        if (threadChannel.getHistory()
+            .getRetrievedHistory()
```

Comment:
Has this been tested? I don't think this works?

Reply:
yeah. u can check the thread activity indicator feature to see how to properly retrieve history.

also, u probably dont want to retrieve "all history", as that could potentially take longer. instead, just do it like the activity indicator. retrieve a history of 10 messages and then see if OP posted anything there.

Reply:
Just retrieve 1 message

Reply:
thats not enough since that can be a bot message or a message from another user, a helper for example.

u cant tell it to "retrieve the first message by author x" without retrieving possibly everything. so u retrieve maybe 10 messages and if none of them is from OP, u conclude that OP didnt say anything yet

Reply:
In that case could we check for the number of messages in the message history? We could have some sort of algorithm to reduce false positives

Reply:
whats the issue with just doing what i said? retrieve 10 messages, check if one of them is from OP. if not, OP likely didnt say anything yet. should cover 98% of use cases.

if it makes u feel better, do 20 instead of 10, i guess 🤷

Reply:
Done